src/components/definition-list/dt.component.js

```diff
@@ -1,11 +1,19 @@
-import React from "react";
+import React, { useContext } from "react";
 import propTypes from "@styled-system/prop-types";
 import PropTypes from "prop-types";
 import { StyledDt } from "./definition-list.style";
+import DlContext from "./__internal__/dl.context";
+
+const Dt = ({ children, ...rest }) => {
+  const { asSingleColumn } = useContext(DlContext);
 
-const Dt = ({ pr = 3, children, mb = 2, ...rest }) => {
   return (
-    <StyledDt data-element="dt" mb={mb} pr={pr} {...rest}>
+    <StyledDt
+      data-element="dt"
+      mb={rest.mb || asSingleColumn ? undefined : 2}
+      pr={rest.pr || asSingleColumn ? undefined : 3}
```

Comment:
I'm not sure, but I think it would be nice to separate out the `mb` and `pr` props there it will be more readable and props won't be overwritten by `{...rest}`.

Reply:
I could destructure them within the component itself but having it done in the args causes the same issues with the prop tables I mentioned above

![image](https://user-images.githubusercontent.com/44157880/131467788-b8339f33-86bb-4e3a-ac55-072d115efde5.png)